File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/resource/CloudStackRequest.java
package com.sequenceiq.cloudbreak.cloud.event.resource;

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformRequest;
import com.sequenceiq.cloudbreak.cloud.event.context.CloudContext;
import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
import com.sequenceiq.cloudbreak.cloud.model.CloudStack;

public class CloudStackRequest<T> extends CloudPlatformRequest<T> {

    private CloudStack cloudStack;

    public CloudStackRequest(CloudContext cloudContext, CloudCredential cloudCredential, CloudStack cloudStack) {
        super(cloudContext, cloudCredential);
        this.cloudStack = cloudStack;
    }

    public CloudStack getCloudStack() {
        return cloudStack;
    }

    @Override
    public String toString() {
        return "CloudStackRequest{" +
                ", cloudStack=" + cloudStack +
                '}';
    }
}

File: cloud-reactor-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/resource/CloudStackRequest.kt
package com.sequenceiq.cloudbreak.cloud.event.resource

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformRequest
import com.sequenceiq.cloudbreak.cloud.context.CloudContext
import com.sequenceiq.cloudbreak.cloud.model.CloudCredential
import com.sequenceiq.cloudbreak.cloud.model.CloudStack

open class CloudStackRequest<T>(cloudContext: CloudContext, cloudCredential: CloudCredential, val cloudStack: CloudStack) : CloudPlatformRequest<T>(cloudContext, cloudCredential) {

    override fun toString(): String {
        return "CloudStackRequest{"
        +", cloudStack=" + cloudStack
        +'}'
    }
}
